tests/test_functional.py

```diff
@@ -95,15 +104,17 @@ def get_tests():
 def test_functional(test_file):
     LintTest = (
         LintModuleOutputUpdate(test_file)
-        if UPDATE
+        if UPDATE.exists()
         else testutils.LintModuleTest(test_file)
     )
     LintTest.setUp()
     LintTest._runTest()
 
 
 if __name__ == "__main__":
-    if "-u" in sys.argv:
-        UPDATE = True
-        sys.argv.remove("-u")
+    if testutils.UPDATE_OPTION in sys.argv:
+        UPDATE.touch()
```

Comment:
I don't see the point in using a file here. Could you explain why?

Reply:
The scope is not the same than for the file so a variable won't work (UPDATE =True modified the UPDATE variable local to `if __name__ == "main":` and not the file's UPDATE variable) . We also can't pass arguments to pytest.main() or we get a parsing error, I think the "clean" solution would be to create a pytest plugin but it's a lot of work.

Reply:
I understand for the scope. Maybe we could have deal with `global` command but i don't know if it is a better idea.
However i don't understand the point of passing arguments to `pytest.main()` because i don't see anywhere usage of `pylint-functional-test-update` file.

Reply:
Yes, globals could work too if you don't like the file thing. Regarding the argument passing to pytest I commented the part of the code where it would be used. If we create a pylint plugin we could make it work from anywhere with `pytest .` and not just this script, but it's a lot of work for a very small simplification. I might do it if I want to learn pytest's plugins later on.

Reply:
Thanks for these explanations @Pierre-Sassoulas . However i don't feel comfortable with the creation of a file just to act as a boolean flag. In fact the file in itself is never used, it is just its existence, so just a boolean condition.
Maybe we can ask for an advice of @AWhetter .

Reply:
OK so i tested global and it does not work. I think it's because we're launching another process entirely with `pytest.main()`, and we need something to communicate with pytest.